src/exo/API_scheduling.py

```diff
@@ -902,6 +902,35 @@ def commute_expr(proc, expr_cursors):
     return Procedure(ir, _provenance_eq_Procedure=proc, _forward=fwd)
 
 
+@sched_op([ExprCursorA])
+def reassociate_expr(proc, expr):
+    """
+    Reassociate the binary operations of '+' and '*'.
+
+    args:
+        expr - the expression to reassociate
+
+    rewrite:
+        a + (b + c)
+            ->
+        (a + b) + c
+    """
+    expr = expr._impl
+    if not isinstance(expr._node, LoopIR.BinOp) or (
+        expr._node.op != "+" and expr._node.op != "*"
+    ):
+        raise TypeError(f"Only '+' or '*' can be reassociated, got {expr._node.op}")
+    if (
+        not isinstance(expr._node.rhs, LoopIR.BinOp)
```

Comment:
Wouldn't it make sense to reassociate LHS `(a + b) + c` as well?

Reply:
Reassociate `(a + b) +c` to `a + (b + c)` can be done this way:
`(a + b) +c` --commute--> `c + (a + b)` --commute--> `c + (b + a)` --reassociate--> `(c + b) + a` --commute--> ` a + (c + b)` --commute--> ` a + (b + c)`

We can provide such wrapper in the standard library.

Reply:
I guess that'll be ambiguos for some cases e.g., `(a + b) + (c + d)`, it is not clear how it will be associated if LHS is also supported.

Reply:
I don't follow your statement. Can you clarify?

Reply:
I was trying to say that hardcoding the direction of reassociation (as you do rn) is correct instead of supporting LHS reassociation in the same primitive, because reassociating `(a + b) + (c + d)` might result in `(a) + (b + (c + d))` or `((a + b) + c) + (d)` depending on the direction, which is ambiguous.